docs/design/cloud-init.rst

```diff
@@ -0,0 +1,353 @@
+
+.. code-block:: yaml
+
+  swap:
+    filename: <file>
+    size: <"auto"/size in bytes>
+    maxsize: <size in bytes>
+
+With this key we can add a swap file but not a swap partition. Since cloud-init
+does the partitioning itself and using swap files is as efficient as using
+partitions, in case the *SWAP* image property is defined in the first form,
+it's better if we just ignore the partition id and use the swap config key to
+create a swap file of the requested size instead.
+
+In case the *SWAP* property is defined in the second form and a swap disk is
+requested, we could make use of the ``mounts`` key of the ``cc_mounts`` module,
+to put the appropriate entry in fstab:
+
+
+.. code-block:: yaml
+
+  mounts:
+    - [ /dev/ephemeral-1, /mnt, auto, "defaults,noexec" ]
+    - [ sdc, /opt/data ]
+    - [ xvdh, /opt/data, "auto", "defaults,nofail", "0", "0" ]
+
+Unfortunately, this module does not support providing non ephemeral device
+names (UUID for swap) and using the standard device naming is error-prone.
+Hence, for swap disks, its better if we bypass cloud-init and let snf-image
+directly format the disk and put the swap entry to fstab.
+
+**50AssignHostname**: Adding a new ``local-hostname`` key in the metadata file
+should be enough to set the hostname. Alternatively, we could make use of the
+``cc_update_hostname`` module which supports the following keys:
+
+.. code-block:: yaml
+
+  preserve_hostname: <true/false>
+  fqdn: <fqdn>
+  hostname: <fqdn/hostname>
+
+We could ignore the fqdn key and use the other two.
+
+**50ChangeMachineId**: We should probably leave this intact. Newer cloud-init
+version will automatically change the machine ID to a random value as this task
+does, but allowing this task to run to make sure the machine ID is always
+altered even on images that user older versions of cloud-init won't harm.
+
+**50ChangePassword**: This task will change the password and inject SSH
+authorized keys to a list of users defined in the ``USERS`` image property. For
+changing the password of users, we cloud make use of the ``cc_set_password``
+module:
+
+.. code-block:: yaml
+
+    ssh_pwauth: <yes/no/unchanged>
+
+    password: password1
+    chpasswd:
+        expire: <true/false>
+
+    chpasswd:
+        list: |
+            user1:password1
+            user2:RANDOM
+            user3:password3
+            user4:R
+
+The ``password`` key only works for the default user and is not present in
+older versions of cloud-init, which leaves us with the ``chpasswd``. Using this
+key we can define the list of user-password tuples.
+
+Injecting SSH authorized keys to a list of users is not that easy. We can make
+the keys available to cloud-init by either setting the ``public-keys`` metadata
+key or using the ``ssh_authorized_keys`` config key. The ``cc_ssh`` module will
+inject the keys found there to root as well as the default user if defined.
+
+keys In order to make cloud-init work with the
```

Comment:
Yes and actually the whole paragraph is left from the previous update and needs rephrasing.